src/java/org/apache/cassandra/utils/ByteBufferUtil.java

```diff
@@ -866,4 +869,27 @@ private static boolean startsWith(ByteBuffer src, ByteBuffer prefix, int offset)
 
         return true;
     }
-}
+
+    /**
+     * Returns true if the buffer at the current position in the input matches given buffer.
+     * If true, the input is positioned at the end of the consumed buffer.
+     * If false, the position of the input is undefined.
+     * <p>
+     * The matched buffer is unchanged
+     *
+     * @throws IOException
+     */
+    public static boolean equalsWithShortLength(FileDataInput in, ByteBuffer toMatch) throws IOException
```

Comment:
nit: Untested codepath?

Reply:
The `test-trie` target would be the one exercising it.

Added a short direct test.